File: python/problem027.py
import math

number_array = [True] * 100000

for i in range(2, int(math.sqrt(100000))):
    if number_array[i]:
        j = i + i
        while j < 100000:
            number_array[j] = False
            j += i

primes_under_thousand = []
for i in range(2, 1000):
    if number_array[i]:
        primes_under_thousand.append(i)

max_count = 0
max_ab = 0
for b in primes_under_thousand:
    for a in range(2 - b, 1000):
        count = 0
        n = 0
        while n * n + a * n + b > 0 and number_array[n * n + a * n + b]:
            count += 1
            if count > max_count:
                max_count = count
                max_ab = a * b
            n += 1

result = max_ab
print(result)

File: javascript/problem027.js
var array = new Array(100000);

for(i=2; i<Math.sqrt(100000); i++){
  if(array[i] == undefined){
    for(j=i+i; j<100000; j = j + i){
      array[j] = false;
    }
  }
}

var primesUnderThousand = [];
for(i=2; i<1000; i++){
  if(array[i] == undefined){
    primesUnderThousand.push(i);
  }
}

var maxCount = 0;
var maxab;
for(i in primesUnderThousand){
  var b = primesUnderThousand[i];
  for(a = 2 - b; a<1000; a++){
    var count = 0;
    var n = 0;
    while(n*n + a*n + b > 0 && array[n*n + a*n + b] == undefined){
      count = count + 1;
      if(count > maxCount){
        maxCount = count;
        maxab = a * b;
      }
      n = n + 1;
    }
  }
}
var result = maxab;
console.log(result);
